proxy/proxy/client.go

```diff
@@ -377,7 +377,7 @@ func needsRefresh(e cacheEntry, refreshCfgBuffer time.Duration) bool {
 	if e.done == nil { // no refresh started
 		return true
 	}
-	if !isValid(e) || e.cfg.Certificates[0].Leaf.NotAfter.Sub(time.Now()) <= refreshCfgBuffer {
+	if !isValid(e) || e.cfg.Certificates[0].Leaf.NotAfter.Round(0).Sub(time.Now().Round(0)) <= refreshCfgBuffer {
```

Comment:
nit: Do we need to strip out of both? Shouldn't just 1 be enough?

Reply:
I think we want to strip off the monotonic reading from both values if I'm reading this correctly: https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/time/time.go;l=848